app/web/src/features/messages/requests/MarkAllReadButton.tsx

```diff
@@ -0,0 +1,66 @@
+import { Tooltip } from "@material-ui/core";
+import IconButton from "components/IconButton";
+import { DoneAllIcon } from "components/Icons";
+import { MARK_ALL_READ } from "features/messages/constants";
+import { useMutation } from "react-query";
+import { service } from "service";
+import getAllPages from "utils/getAllPages";
+
+export default function MarkAllReadButton({
+  type,
+}: {
+  type: "chats" | "hosting" | "surfing";
+}) {
+  const markAll = useMutation(async () => {
+    if (type === "chats") {
+      const data = await getAllPages({
+        serviceFunction: service.conversations.listGroupChats,
+        listKey: "groupChatsList",
+        params: (previousData) => previousData?.lastMessageId,
+        hasMore: (previousData) => !previousData.noMore,
+      });
+      await Promise.all(
+        data.map<void>((chat) =>
+          chat.latestMessage
```

Comment:
Is `latestMessage` meant to be `null` / `undefined` when all the messages in a chat is marked as read? It doesn't seem to be working on the preview site, as it fires a tonne of `MarkLastSeenGroupChat` every time I click the button even though it's all read on the test account.

Maybe it's more reliable to use `unseenMessageCount` instead?

Reply:
No there's no checks but I'll do that